src/v/cloud_roles/azure_aks_refresh_impl.cc

```diff
@@ -0,0 +1,195 @@
+    // &grant_type=client_credentials
+
+    auto jwt_token_fut = co_await ss::coroutine::as_future(
+      read_fully_to_string({federated_token_file_}));
+    if (jwt_token_fut.failed()) {
+        vlog(
+          clrl_log.error,
+          "failed to read OIDC pod token from file {}",
+          federated_token_file_);
+        co_return ss::coroutine::exception(jwt_token_fut.get_exception());
+    }
+
+    auto jwt_token = std::string{jwt_token_fut.get()};
+    boost::trim(jwt_token);
+
+    // setup header for www-form-urlencoded body
+    auto access_token_req
+      = boost::beast::http::request<boost::beast::http::string_body>{};
+    access_token_req.method(boost::beast::http::verb::post);
+    access_token_req.target(fmt::format("/{}/oauth2/v2.0/token", tenant_id_));
+    access_token_req.set(
+      boost::beast::http::field::host,
+      fmt::format("{}:{}", address().host(), address().port())),
+      access_token_req.set(
+        boost::beast::http::field::user_agent, "redpanda.vectorized.io");
+    access_token_req.set(
+      boost::beast::http::field::content_type,
+      "application/x-www-form-urlencoded");
+
+    // compose the body of the request with the informations retrieved from the
+    // env variables
+    auto body = ssx::sformat(
+      "scope={}"
+      "&client_id={}"
+      "&client_assertion_type=urn%3Aietf%3Aparams%3Aoauth%3Aclient-assertion-"
+      "type%3Ajwt-bearer"
+      "&client_assertion={}"
+      "&grant_type=client_credentials",
+      uri_encode("https://storage.azure.com/.default", true),
+      client_id_,
+      jwt_token);
+
+    // TODO check
+    // AKS requires a TLS enabled client by default, but in test mode where we
+    // use the http imposter, we need to use a simple client.
+    auto tls_enabled = refresh_credentials::client_tls_enabled::yes;
+    if (address().port() != default_port) {
+        tls_enabled = refresh_credentials::client_tls_enabled::no;
+    }
+
+    co_return co_await request_with_payload(
+      co_await make_api_client("azure_aks_oidc", tls_enabled),
+      std::move(access_token_req),
+      std::move(body));
+}
+
+auto azure_aks_refresh_impl::parse_response(iobuf resp)
+  -> api_response_parse_result {
+    // schema for (an example)
+    //  {
+    //    "token_type": "Bearer",
+    //    "expires_in": 3599,
+    //    "access_token":
+    //    "eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiIsIng1dCI6Ik1uQ19WWmNBVGZNNXBP..."
+    //  }
+    constexpr static auto success_schema_str = R"json(
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "type": "object",
+  "properties": {
+    "token_type": {
+      "type": "string",
+      "pattern": "Bearer"
+    },
+    "expires_in": {
+      "type": "integer"
+    },
+    "access_token": {
+      "type": "string"
+    }
+  },
+  "required": [
+    "token_type", 
+    "expires_in",
+    "access_token"
+  ]
+}
+)json";
+
+    vlog(clrl_log.error, "{}", resp.hexdump(20000000));
```

Comment:
was this log line intentional? would this dump up to 20 MB? does it contain any potential sensitive information.

Reply:
a typo